### PR TITLE
Removes default clear button in date input type

### PR DESCRIPTION
**Pull request by kapilgarg1996**
Fixes issue <a href="https://github.com/jquery/jquery-mobile/issues/8001">#8001</a>

### CODE REVIEW

**Comment by arschmitz** [comment]
@kapilgarg1996 thank you for the PR. While your fix is correct based on what we are currently doing i think we should change how this works.

Right now we are just always hiding the clear button regardless of if we are adding one leaving no way to use the native if you want.

I think we should change the current rule being modified to 

```css
.ui-textinput-hide-clear::-ms-clear,
.ui-textinput-hide-clear::-ms-clear {
 	display: none;
 }
.ui-textinput-hide-clear::-webkit-clear-button {
 	display: none;
 }
```

And in the clear button extension we should add the class `ui-textinput-hide-clear` to the input  when we add a clear button. 

This way if a clear button is added by jqm we hide the native clear. If no clear button is added we leave the default native if available. If you want no clear button at all native or jqm then you can simply add `.ui-textinput-hide-clear` to the input your self and make sure `clearBtn` option is `false` ( the default ) .

@gabrielschulhof @jaspermdegroot  @cgack thoughts?

**Comment by kapilgarg1996** [comment]
@arschmitz  Yeah that'll work perfectly.

**Comment by gabrielschulhof** [comment]
@arschmitz I think that's a good idea.

File: css/structure/jquery.mobile.forms.textinput.css [hunk 1]
```diff
@@ -114,11 +114,14 @@ for Android 4.1 */
 .ui-input-text input[type=number]::-webkit-outer-spin-button {
 	margin: 0;
 }
-/* Resolves issue #5756: Textinput in IE10 has a default clear button */
-.ui-input-text input::-ms-clear,
-.ui-input-search input::-ms-clear {
+/* Resolves issue #8001: Default clear button in date input type*/
+.ui-hide-default-clear::-ms-clear,
```
Comment (sfrisk) [inline css/structure/jquery.mobile.forms.textinput.css]: remove coma

File: css/structure/jquery.mobile.forms.textinput.css [hunk 2]
```diff
@@ -114,11 +114,14 @@ for Android 4.1 */
...
+ {
 	display: none;
 }
+.ui-hide-default-clear::-webkit-clear-button{
+display: none;
```
Comment (sfrisk) [inline css/structure/jquery.mobile.forms.textinput.css]: The `display: none` should be indented. 
http://contribute.jquery.org/style-guide/css/#spacing

**Comment by jaspermdegroot** [comment]
@arschmitz +1

Re: "/* This needs to be its own rule or it does not work in chrome for some reason */".
That is because browsers have to drop the whole rule if they encounter an invalid pseudo element and all vendor prefixed pseudo elements are invalid for other vendors.

**Comment by arschmitz** [comment]
> That is because browsers have to drop the whole rule if they encounter an invalid pseudo element and all vendor prefixed pseudo elements are invalid for other vendors.

@jaspermdegroot I didn't know that good to know thank you @kapilgarg1996  lets update the comment

**Comment by kapilgarg1996** [comment]
@jaspermdegroot Thanks for the information.
@arschmitz comment updated. have a look at it.
@sfrisk Thanks sarah for the styling suggestions.

File: js/widgets/forms/clearButton.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ define( [
 		},
 
 		_addClearBtn: function() {
-
+			this.widget().children('input').addClass("ui-hide-default-clear") ; /*Removes default clear button if jqm clear button is ON*/
```
Comment (gabrielschulhof) [inline js/widgets/forms/clearButton.js]: @kapilgarg1996 
- please use double quotes for strings
- please put single spaces around strings, i.e., ```( "abc" )```
- please place the comment on the previous line, leaving a blank line before it, and use ```//``` as the comment delimiter

Comment (kapilgarg1996) [inline js/widgets/forms/clearButton.js]: @gabrielschulhof done :+1:

**Comment by arschmitz** [comment]
This looks good now will land shortly

**Pull request closed without merging**